community/values/src/main/java/org/neo4j/values/storable/DurationValue.java

```diff
@@ -192,10 +192,20 @@ DurationValue create(
 
     private static final DurationValue ZERO = new DurationValue( 0, 0, 0, 0 );
     private static final List<TemporalUnit> UNITS = unmodifiableList( asList( MONTHS, DAYS, SECONDS, NANOS ) );
+    // This comparator is safe until 292,271,023,045 years. After that, we have an overflow.
+    private static final Comparator<DurationValue> COMPARATOR = Comparator.comparingLong(DurationValue::averageLengthInSeconds)
+            // nanos are guaranteed to be smaller than NANOS_PER_SECOND
+            .thenComparingLong( (d) -> d.nanos )
+            // At this point, the durations have the same length and we compare by the individual fields.
+            .thenComparingLong( (d) -> d.months )
+            .thenComparingLong( (d) -> d.days )
+            .thenComparingLong( (d) -> d.seconds );
+
     static final int NANOS_PER_SECOND = 1_000_000_000;
     static final long SECONDS_PER_DAY = DAYS.getDuration().getSeconds();
     /** 30.4375 days = 30 days, 10 hours, 30 minutes */
     private static final double AVERAGE_DAYS_PER_MONTH = 365.25 / 12;
+    static final long AVG_SECONDS_PER_MONTH = 2_629_800;
```

Comment:
Actually `2_629_746` if we want to be really exact...

Reply:
The smallest duration in months and years for which there is an exact number of days is `400 years` = `((365*4+1)*25-1)*4+1 days` = `146_097 days` which is `12_622_780_800 seconds` (discounting leap seconds, which our implementation does). Divide that by `400*12` to get the average number of seconds in a month: `2_629_746`

Reply:
Yes. Since the rest of the implementation is based on 365.25 days per year, I used this value instead.